### PR TITLE
WIP: Apache webserver handles TLS on Ironic and Ironic-inspector sides

### PR DESCRIPTION
**Pull request by namnx228**
This PR adds an option to let Apache web server handle TLS on Ironic and Ironic-inspector side.
Related PRs: 
https://github.com/metal3-io/ironic-image/pull/230
https://github.com/metal3-io/baremetal-operator/pull/728
metal3-io/ironic-inspector-image#70

### CODE REVIEW

File: lib/common.sh [hunk 1]
```diff
@@ -149,6 +149,9 @@ export IRONIC_INSPECTOR_IMAGE=${IRONIC_INSPECTOR_IMAGE:-"quay.io/metal3-io/ironi
 export IRONIC_DATA_DIR="$WORKING_DIR/ironic"
 export IRONIC_IMAGE_DIR="$IRONIC_DATA_DIR/html/images"
 export IRONIC_KEEPALIVED_IMAGE=${IRONIC_KEEPALIVED_IMAGE:-"quay.io/metal3-io/keepalived"}
+#TODO: Update vars.md
```
Comment (fmuyassarov) [inline lib/common.sh:152]: nit: 
```suggestion
```

**Review by fmuyassarov — COMMENTED** [review]
/lgtm

**Review by smoshiur1237 — COMMENTED** [review]
Lets pass the CI. Otherwise Looks good to me.

File: lib/common.sh [hunk 2]
```diff
@@ -159,6 +159,12 @@ export IRONIC_INSPECTOR_IMAGE=${IRONIC_INSPECTOR_IMAGE:-"quay.io/metal3-io/ironi
 export IRONIC_DATA_DIR="$WORKING_DIR/ironic"
 export IRONIC_IMAGE_DIR="$IRONIC_DATA_DIR/html/images"
 export IRONIC_KEEPALIVED_IMAGE=${IRONIC_KEEPALIVED_IMAGE:-"quay.io/metal3-io/keepalived"}
+# Option to use reverse proxy on ironic-inspector
+export INSPECTOR_REVERSE_PROXY_SETUP=${INSPECTOR_REVERSE_PROXY_SETUP:-"false"}
```
Comment (furkatgofurov7) [inline lib/common.sh:163]: I might be missing the bits, but are not we supposed to have this option enabled by default? AFAIK, `IRONIC_TLS_SETUP` is used by default.
Also, vars.md needs to be updated with new var.

**Review by furkatgofurov7 — COMMENTED** [review]
Question inline:

File: lib/common.sh (hunk 2 above)
Comment (furkatgofurov7) [reply lib/common.sh:163]: I was not fully clear in prev. comment, maybe. Is it the case that we will have 2 options depending on `IRONIC_TLS_SETUP`?:

-  `IRONIC_TLS_SETUP=false` ==> `INSPECTOR_REVERSE_PROXY_SETUP=false`;

- `IRONIC_TLS_SETUP=true` ==> `INSPECTOR_REVERSE_PROXY_SETUP=false/true`;

**Review by fmuyassarov — COMMENTED** [review]
LGTM, squashing the commits would be good I think.

**Comment by fmuyassarov** [comment]
Sorry I forgot to mention, you probably need to add the image for upgrade tests as well https://github.com/metal3-io/metal3-dev-env/blob/a081567ce31585b7276d35783cb1180792955200/scripts/feature_tests/upgrade/combined_tests/1cp_1w_bootDiskImageANDK8sControllers_clusterLevel_upgrade.sh#L93

**Comment by kashifest** [comment]
@namnx228 Whats the update with this PR?

**Comment by namnx228** [comment]
I think we can close this PR. It is not needed anymore.

**Pull request closed without merging**